integration_test/agents/agents.go

```diff
@@ -813,9 +868,11 @@ func SetupOpsAgentFrom(ctx context.Context, logger *log.Logger, vm *gce.VM, conf
 		if err := gce.UploadContent(ctx, logger, vm, strings.NewReader(config), util.GetConfigPath(vm.ImageSpec)); err != nil {
 			return fmt.Errorf("SetupOpsAgentFrom() failed to upload config file: %v", err)
 		}
-		if err := restartOpsAgent(ctx, logger, vm); err != nil {
+		if err := RestartOpsAgent(ctx, logger, vm); err != nil {
 			return err
 		}
+	} else if IsOpsAgentUAPPlugin() {
+		return RestartOpsAgent(ctx, logger, vm)
```

Comment:
Why do we need to restart the ops agent plugin here specifically?

Reply:
Not like the regular ops agent, the agent plugin is not `started` automatically after installation. It needs to be explicitly started. 

The point of calling`RestartOpsAgent` function is to trigger a `start` of the agent. 

When len(config)> 0, the regular agent needs to be re-started, and the plugin agent needs to be started to apply the custom config. 

When len(config) ==0, only the plugin agent needs to be started.

Reply:
In that case, is it possible to be more explicit here and call `StartOpsAgentPlugin` instead?

Reply:
`StartOpsAgentPlugin` and `RestartOpsAgent` do different things for the agent plugin; `StartOpsAgentPlugin` starts the plugin grpc server itself. However, `RestartOpsAgent` sends a gRPC request to the plugin's `Start()` endpoint, to start the Ops Agent. 
In this case the plugin gRPC server is live already, I only need to start the Ops Agent.